### PR TITLE
[ADAM-1832] Use awesome list style and link to bigdatagenomics/awesome-adam.

### LINKED ISSUE #1832: Add link to awesome list of applications that extend ADAM
[Awesome List](https://github.com/sindresorhus/awesome)s are a lightweight way to link together curated lists on Github.  ADAM is already on the [Apache Spark awesome list](https://github.com/awesome-spark/awesome-spark).

Would it be easier for folks to contribute to such a list in an external repository as compared to pull requesting here against the readme?

Templated such an external repository here
https://github.com/heuermh/awesome-adam

I'd suggest updating the link format of our readme and docs (to use the `-` and full sentence description) and then link out, something like

For additional resources, see the [awesome list of applications that extend ADAM](https://github.com/heuermh/awesome-adam).

### PR DESCRIPTION
**Pull request by heuermh**
Fixes #1832.

Would require migrating https://github.com/heuermh/awesome-adam to bigdatagenomics organization.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -96,23 +96,25 @@ platforms.
...
-  [GA4GH schemas](https://github.com/ga4gh/schemas)
+* [DECA](https://github.com/bigdatagenomics/deca) - DECA is a reimplementation of the
+  XHMM copy number variant caller on top of ADAM.
+* [Gnocchi](https://github.com/bigdatagenomics/gnocchi) - Gnocchi provides primitives
+  for running GWAS/eQTL tests on large genotype/phenotype datasets using ADAM.
+* [Lime](https://github.com/bigdatagenomics/lime) - Lime provides a
+  parallel implementation of genomic set theoretic primitives using the ADAM
+  [region join](http://adam.readthedocs.io/en/latest/api/joins/) API.
+* [Mango](https://github.com/bigdatagenomics/mango) - Mango is a library for
+  visualizing large scale genomics data with interactive latencies.
```
Comment (fnothaft) [inline README.md]: Maybe we could add some of your adam examples and some of @antonkulaga's codes? Also @ryan-williams?

Comment (heuermh) [reply README.md]: The idea is that it is easier to pull request against the separate awesome-adam repo than pull request here.

Comment (fnothaft) [reply README.md]: Oh, sorry about that, I totally missed that. Yes, you are correct!

**Comment by heuermh** [comment]
Yup, just did so.  I'll rebase this pull request with that change in a sec.

**Comment by fnothaft** [comment]
Merged! Thanks @heuermh!

**Pull request merged**